src/logged-evaluation.nix

```diff
@@ -32,16 +33,16 @@ let
   # references to all the store paths needed, preventing the shell's
   # actual environment from being deleted.
   wrapped-project = shell: services: derivation (
-    shell.drvAttrs // {
-      name = "lorri-wrapped-project-${shell.name}";
+    (shell.drvAttrs or {}) // {
+      name = "lorri-wrapped-project-${shell.name or "unknown"}";
 
       origExtraClosure = shell.extraClosure or [];
       extraClosure = runtimeCfg.closure;
 
-      origBuilder = shell.builder;
+      origBuilder = shell.builder or null;
       builder = runtimeCfg.builder;
 
-      origSystem = shell.system;
+      origSystem = shell.system or null;
```

Comment:
Having to introduce so many optional fields means we have a leaky abstraction.

Reply:
I've attempted to address this here: https://github.com/target/lorri/pull/266